backend/src/main/java/atlantbh/restaurants/controllers/UserController.java

```diff
@@ -0,0 +1,26 @@
+package atlantbh.restaurants.controllers;
+
+import atlantbh.restaurants.models.dto.UserRegistrationDTO;
+import atlantbh.restaurants.models.User;
+import atlantbh.restaurants.services.UserService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+@RestController
+public class UserController extends BaseController<User, UserService> {
+
+    public ResponseEntity registerUser(@Valid @RequestBody UserRegistrationDTO user, BindingResult bindingResult) throws Exception {
+        try {
+            User registeredUser = service.get(user, bindingResult);
```

Comment:
You should not pass bindingResult to service.. That error handling should be in controller..
```
if (bindingResult.hasErrors()) {
           return badRequest
        }
```

Reply:
Fixed c2ed9b4060983e1312e68ee9cdd1de2be1986207